src/transformations/utils/build-super-calls-for-chain.ts

```diff
@@ -110,6 +109,32 @@ function isImplicitlyConstructed(contract: ContractDefinition): boolean {
 
   return (
     contract.contractKind === 'contract' &&
-    (ctor == undefined || ctor.parameters.parameters.length === 0)
+    (ctor == undefined ||
+      ctor.parameters.parameters.length === 0 )
+  );
+}
+
+function isEmpty(contract: ContractDefinition): boolean {
+  const ctor = getConstructor(contract);
+  const varInitNodes = [...findAll('VariableDeclaration', contract)].filter(
+    v => v.stateVariable && v.value && !v.constant && !hasOverride(v, 'state-variable-assignment'),
+  );
+  const modifiers = [];
+  if(ctor){
+    // we only include modifiers that don't reference base contracts
+    for (const call of ctor.modifiers) {
+      const { referencedDeclaration } = call.modifierName;
+      if (referencedDeclaration != null && !ctor.linearizedBaseContracts.includes(referencedDeclaration)) {//is not inheritance
```

Comment:
I don't understand the `!ctor.linearizedBaseContracts.includes(referencedDeclaration))` part.

what does it mean for a modifier to be in the linearizedBaseContracts ?

Reply:
When the ast filters by modifier it takes both modifiers, as in `modifier hasmodifier(){_;}`, and base contract contructors, so in this case `constructor (string name, string symbol) ERC20(name, symbol) hasmodifer` the `.modifiers` will bring both as modifiers, but for our check we only need to know if those are actual modifiers.

This was better addressed on the last commit.